superset-frontend/src/SqlLab/components/HighlightedSql/HighlightedSql.test.tsx

```diff
@@ -0,0 +1,48 @@
+/**
```

Comment:
The code is generally well-written and follows good practices. However, there are a few minor improvements that could be made:

1. **Test Descriptions:** Make your test descriptions more descriptive to better understand what each test is doing. For example, instead of 'renders with props', you could say 'renders HighlightedSql component with sql prop'.

2. **Consistent Quotation Marks:** In the sql constant declaration, you are using double quotes for the SQL string. It would be better to stick to single quotes for consistency.

3. **Test Variable Names:** In your last test, the variable name 'syntaxHighlighter' is a bit misleading as it's actually a collection of 'code' elements, not 'SyntaxHighlighter' components. Consider renaming it to something like 'codeElements'.

Here's how the improved code could look:

```jsx
const sql = 'SELECT * FROM test WHERE something=\'fkldasjfklajdslfkjadlskfjkldasjfkladsjfkdjsa\'';
test('renders HighlightedSql component with sql prop', () => {
  expect(React.isValidElement(<HighlightedSql sql={sql} />)).toBe(true);
});
test('renders a ModalTrigger component', () => {
  const { getByTestId } = render(<HighlightedSql sql={sql} />);
  expect(getByTestId('span-modal-trigger')).toBeInTheDocument();
});
test('renders a ModalTrigger component with shrink prop and maxWidth prop set to 20', () => {
  const { getByTestId } = render(
    <HighlightedSql sql={sql} shrink maxWidth={20} />,
  );
  expect(getByTestId('span-modal-trigger')).toBeInTheDocument();
});
test('renders two code elements in modal when rawSql prop is provided', () => {
  const { getByRole, queryByRole, getByTestId } = render(
    <HighlightedSql sql={sql} rawSql="SELECT * FORM foo" shrink maxWidth={5} />,
  );
  expect(queryByRole('dialog')).not.toBeInTheDocument();
  fireEvent.click(getByTestId('span-modal-trigger'));
  expect(queryByRole('dialog')).toBeInTheDocument();
  const codeElements = getByRole('dialog').getElementsByTagName('code');
  expect(codeElements.length).toEqual(2);
});
``

Reply:
Thanks @captain-hoity-toity for the detail improvement. It was basically migrating existing test block to typescript so I was not able to focus on the test quality.

I'll update your improvement suggestion too!